### ISSUE TITLE
OpenTelemetry Tracing Support

### ISSUE THREAD

**Issue by endigma** [golangci/golangci-lint#4038]
### Your feature request related to a problem? Please describe.

We have a large project that takes a very long time to lint, we'd like to figure out which linters take the longest so we can reconsider their usage or how they can be applied.

### Describe the solution you'd like.

I'd like support for OpenTelemetry tracing, so the trace for execution may be viewed in Jaeger, etc.

### Describe alternatives you've considered.

There is baked in support for go's tracing tools, however these aren't very good at handling massive amounts of traces and they record tons of information that doesn't really matter to us. It also doesn't support shipping traces over the network (we'd like to send traces to a remote OTLP server for CI analytics). Basic OpenTelemetry Tracing support would allow us to view less specific performance information over time.

**Comment by Antonboom** [comment]
`golangci-lint run` already has verbose mode (`-v`) and supports `GL_DEBUG` env, e.g.

```bash
GL_DEBUG=exec golangci-lint run -v ./...
```

It looks like we don't need introduce OpenTel dependency (and the tool is questionable for these purposes) but just add more info/debug logs.

**Comment by endigma** [comment]
Well, it _is telemetry_ and I think it being open would be best? I'd specifically like to record the timings over time, which logging (especially unstructured logging) does not help with.

**Comment by ldez** [comment]
The timing is only for debugging, I think logs are enough and better for the context of golangci-lint.
